src/main.rs

```diff
@@ -244,13 +249,35 @@ async fn main() {
         let leader_hpke_config: HpkeConfig = leader_hpke_config.unwrap();
         let helper_hpke_config: HpkeConfig = helper_hpke_config.unwrap();
 
-        print!("Submitting reports: ({}) ", task.report_count);
+        println!("Submitting reports: ({}) ", task.report_count);
         io::stdout().flush().unwrap();
+        let mut tasks = JoinSet::new();
+
         for i in 0..task.report_count {
-            print!("{} ", i + 1);
-            io::stdout().flush().unwrap();
-            submit_reports_for_task(task, &config, &leader_hpke_config, &helper_hpke_config).await;
+            let client = client.clone();
+            let task = task.clone();
+            let config = config.clone();
+            let leader_hpke_config = leader_hpke_config.clone();
+            let helper_hpke_config = helper_hpke_config.clone();
```

Comment:
@ncloudioj Is this necessary or is it for performance? I would have expected that these can just be borrowed because none of them need to be written to.

Reply:
Oh, we can't pass references (even the shared ones) here because `tokio::task::spawn()` requires the future to be `Send + 'static` (see [here](https://docs.rs/tokio/latest/tokio/task/struct.JoinSet.html#method.spawn)), so we have to use owning arguments instead.